### PR TITLE
fix: change sqla is_relationship for support sqla<=2.0.0

### LINKED ISSUE #2345: ForeignKey column is not listed in view.
from sqlalchemy 2.0.0, It seems foreign key column cannot appear in list-view, edit-view.

Flask-SQLAlchemy==3.0.3
Flask-Admin==1.6.1
sqlalchemy==2.0.6

### LINKED ISSUE #2348: FlaskAdmin not compatible with SQLAlchemy 2.0
FlaskAdmin does not seem compatible with SQLAlchemy 2.0 yet. I'm getting this import error:

```
    from flask_admin.contrib.sqla import ModelView
../../../../../lib/python3.10/site-packages/flask_admin/contrib/sqla/__init__.py:2: in <module>
    from .view import ModelView
../../../../../lib/python3.10/site-packages/flask_admin/contrib/sqla/view.py:18: in <module>
    from flask_admin.contrib.sqla.tools import is_relationship
../../../../../lib/python3.10/site-packages/flask_admin/contrib/sqla/tools.py:11: in <module>
    from sqlalchemy.ext.associationproxy import ASSOCIATION_PROXY
E   ImportError: cannot import name 'ASSOCIATION_PROXY' from 'sqlalchemy.ext.associationproxy' (lib/python3.10/site-packages/sqlalchemy/ext/associationproxy.py)

```

### PR DESCRIPTION
**Pull request by miettal**
close #2345

### CODE REVIEW

**Comment by miettal** [comment]
how about this PR? > @flask-admin maintainer

**Comment by ericvanular** [comment]
Is it possible to merge this PR? Relationships are not fully usable in SQLA2 currently

File: flask_admin/contrib/sqla/tools.py [hunk 1]
```diff
@@ -224,7 +224,8 @@ def is_hybrid_property(model, attr_name):
 
 
 def is_relationship(attr):
-    return hasattr(attr, 'property') and hasattr(attr.property, 'direction')
+    return (hasattr(attr, 'property') and hasattr(attr.property, 'direction')  # sqla<2.0.0
+            or (hasattr(attr, '_is_relationship') and attr._is_relationship))  # sqla>=2.0.0
```
[the remark this answers is not in the document]
Comment (miettal) [reply flask_admin/contrib/sqla/tools.py:228]: It seems this is not documented.
Actually I just found _is_relationship when debugging. this is reason why I use _is_relationship.
if there is a solution that documented (model.registry.configure()), the solution is best!

[the remark this answers is not in the document]
Comment (miettal) [reply flask_admin/contrib/sqla/tools.py:228]: I'm also not familiar with SQLAlchemy, a just user of flask-admin, sqlalchemy. So I follow mantainer's decistion. ;)

**Comment by samuelhwilliams** [comment]
@cjmayo do we still need this for sqlalchemy 1/2 compatibility?

**Pull request closed without merging**